### PR TITLE
Fixed some compilation warnings and removed unused source code

### PR DESCRIPTION
**Pull request by wenqing**
This PR
1) fixes some compilation warnings. 
2) removes unused source code.
3) re-writes the reading of command line arguments in the main function.

### CODE REVIEW

**Comment by norihiro-w** [comment]
it looks ok for me besides one minor comment.

File: Base/FileTools.cpp [hunk 1]
```diff
@@ -155,7 +155,8 @@ std::string getCwd()
 #ifdef WIN32
 	_getcwd(cwd, FILENAME_MAX);
 #else
-	getcwd(cwd, FILENAME_MAX);
+	char* unused __attribute__((unused));
```
[the remark this answers is not in the document]
Comment (wenqing) [reply Base/FileTools.cpp]: @norihiro-w  Thanks. Just removed it and gcc 7.3 still keeps silent after removing.

**Comment by norihiro-w** [comment]
👍 if Jenkins is green

**Comment by wenqing** [comment]
@norihiro-w  Only the two benchmarks failed. The two benchmarks do not work for long time. I will fixed them in another PR.

**Pull request merged**